### PR TITLE
Deactivate scalafix and scalafmt on compile on CI, closes OPS-74

### PR DESCRIPTION
**Pull request by tpetillot**
**Motivation:**
CI must deactivate on compile scalafix and scalafmt parsing

**Modification:**
GatlingAutomatedScalafixPlugin scalafix on compile respect scalafixOnCompile setting

### CODE REVIEW

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala [hunk 1]
```diff
@@ -25,16 +25,24 @@ import sbt._
 import sbt.Keys._
 
 object GatlingAutomatedScalafixPlugin extends AutoPlugin {
+
   override def requires: Plugins = ScalafixPlugin && GatlingBuildConfigPlugin
 
+  override def globalSettings: Seq[Def.Setting[_]] = Seq(
+    scalafixOnCompile := true
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:32]: Where does this setting come from?
We should not redefine a default global setting of a setting created by another plugin.

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala [hunk 2]
```diff
@@ -25,16 +25,24 @@ import sbt._
 import sbt.Keys._
 
 object GatlingAutomatedScalafixPlugin extends AutoPlugin {
+
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:28]: Why?

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala [hunk 3]
```diff
@@ -25,16 +25,24 @@ import sbt._
...
 
     def automateScalafixBeforeCompile(configurations: Configuration*): Seq[Setting[_]] =
       configurations.foldLeft(List.empty[Setting[_]]) {
         _ ++ inConfig(_)(
           Seq(
-            compile := compile.dependsOn(scalafix.toTask("").dependsOn(scalafixWriteConfigFile)).value
+            compile := {
+              if (scalafixOnCompile.value) compile.dependsOn(scalafix.toTask("").dependsOn(scalafixWriteConfigFile)).value
+              else compile.value
+            }
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:45]: When multilined, please use braced branches.

```suggestion
            compile := {
              if (scalafixOnCompile.value) {
                compile.dependsOn(scalafix.toTask("").dependsOn(scalafixWriteConfigFile)).value
              } else {
                compile.value
              }
            }
```

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala (hunk 2 above)
Comment (tpetillot) [reply src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:28]: Same format as GatlingAutomatedScalafmtPlugin

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala (hunk 1 above)
Comment (tpetillot) [reply src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:32]: That's what we do in GatlingAutomatedScalafmtPlugin

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafmtPlugin.scala [hunk 4]
```diff
@@ -40,6 +36,7 @@ object GatlingAutomatedScalafmtPlugin extends AutoPlugin {
   )
 
   override def projectSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmtOnCompile := !sys.env.getOrElse("CI", "false").toBoolean,
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafmtPlugin.scala:39]: Thanks! :bow:

File: src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala (hunk 3 above)
Comment (Isammoc) [reply src/main/scala/io/gatling/build/automated/GatlingAutomatedScalafixPlugin.scala:41]: Forgotten?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -35,3 +35,5 @@ jobs:
           key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/build.properties') }}
       - name: SBT ci-checks
         run: ./sbtx ci-checks
+        env:
+          CI: false
```
Comment (Isammoc) [inline .github/workflows/test.yml:39]: ???
scalafix and scalafmt should be checked (but not at compile time, at check time)

**Pull request merged**